comma/model.py

```diff
@@ -16,12 +20,90 @@ def setup(self, size: int, dir_params: str):
+            "Model parameter(s) not found: %s." % ", ".join(missing_model_params)
+            
+        # check if there is any overlap between lockdown intervals
+        lockdown = params_model["lockdown"]
+        lockdown_intervals = [l["interval"] for l in lockdown] 
+        lockdown_intervals = sorted(lockdown_intervals, key=lambda li: li[0])
+        overlaps = [lockdown_intervals[i][1] >= lockdown_intervals[i+1][0] \
+            for i in range(len(lockdown_intervals) - 1)]
+        assert not any(overlaps), "Lockdown intervals have overlap: \n%s" % "\n".join(
+            ["%s - %s" % (str(lockdown_intervals[i]), str(lockdown_intervals[i+1])) \
+                for i in range(len(overlaps)) if overlaps[i]] 
+        )
+        
+        # check if all the steps are fully covered by the given lockdown intervals
+        steps = params_model["steps"]
+        expected_timeline = list(range(1, steps+1))
+        given_timeline = []
+        for li in lockdown_intervals:
+            given_timeline += list(range(li[0], li[1] + 1))
+        uncovered_steps = set(expected_timeline) - set(given_timeline)
+        assert not uncovered_steps, "Uncovered steps: %s." % ", ".join(
+            [str(us) for us in uncovered_steps]
+        )
+            
+        # check if all hypothesis files exist
+        fnames = ["actions_effects_on_%s.csv" % s for s in params_model["status"]]
+        fnames += ["lockdown_%s.csv" % l for l in params_model["lockdown_policies"]]
+        fpaths = [os.path.join(dir_params, fn) for fn in fnames]
+        fexist = [os.path.isfile(fp) for fp in fpaths]
+        assert all(fexist), "Hypothesis file(s) not found: %s." % ", ".join(
+            [fnames[i] for i in range(len(fnames)) if not fexist[i]]
+        )
+        
+        # check if all hypothesis files contain all the required agent features
+        required_features = ["actions", "baseline"]
+        required_features += _get_one_hot_encoded_features(fpath_params_individual)
+        hypothesis_data = [pd.read_csv(fp, sep=";") for fp in fpaths]
+        missing_features = [set(required_features) - set(hd.columns) for hd in hypothesis_data]
+        assert not any(missing_features), "Missing features:\n%s" % "\n".join(
+            ["%s - %s" % (fnames[i], ", ".join(missing_features[i])) \
+                for i in range(len(fnames)) if missing_features[i]]
+        )
+        
+        # check if all hypothesis files contain hypotheses of all actions
+        required_actions = params_model["actions"]
+        missing_actions = [set(required_actions) - set(hd["actions"]) for hd in hypothesis_data]
+        assert not any(missing_actions), "Missing actions:\n%s" % "\n".join(
+            ["%s - %s" % (fnames[i], ", ".join(missing_actions[i])) \
+                for i in range(len(fnames)) if missing_actions[i]]
+        )
+        
+        # check if any hypothesis file contains out-of-range values
+        required_range = [-1, 1]
```

Comment:
The required range is -1:1 for the effects of actions on mental health, but for number of contacts this is not necessarily true, as these are the number of people increment for each action, so the boundary here can't be defined (e.g., if we want to add at some point an action that allows you to meet 100 people at once, like for example going to a concert we still want to be able to add it without incurring in this error). 

I'd say however, why do we want to check that the range is pre-defined? Wouldn't be best to leave it to the user to assign values that make sense to him/her? these values directly depend on their theoretical assumption about 1) the magnitude of the effects and 2) the directionality